src/irr/asset/CCPUMeshPacker.h

```diff
@@ -0,0 +1,113 @@
+#ifndef __IRR_C_CPU_MESH_PACKER_H_INCLUDED__
+#define __IRR_C_CPU_MESH_PACKER_H_INCLUDED__
+
+#include <irr/asset/ICPUMesh.h>
+#include <irr/asset/IMeshPacker.h>
+
+namespace irr 
+{ 
+namespace asset
+{
+
+using namespace meshPackerUtil;
+
+template <typename MDIStructType = DrawElementsIndirectCommand_t>
+class CCPUMeshPacker final : public IMeshPacker<ICPUMeshBuffer, MDIStructType>
+{
+public:
+	CCPUMeshPacker(const SVertexInputParams& preDefinedLayout, const AllocationParams& allocParams, uint16_t maxIndexCountPerMDIData = std::numeric_limits<uint16_t>::max())
+		:IMeshPacker<ICPUMeshBuffer, MDIStructType>(preDefinedLayout, allocParams, maxIndexCountPerMDIData) 
+	{
+		m_outMDIData = core::make_smart_refctd_ptr<ICPUBuffer>(allocParams.MDIDataBuffSupportedSizeInBytes);
+		m_outIdxBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(allocParams.indexBuffSupportedSizeInBytes);
```

Comment:
create the buffers in a separate method which needs to be called before the first `commit` (you can check and assert on that)

Aside from that the CPU MeshPacker should (or even the GPU in the future) posses a `shrink_to_fit` method that shrinks the address space (for now, assert and forbid shrinking if the CPU buffers have already been created)